include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp

```diff
@@ -498,6 +551,124 @@ struct touch : public base_turn_handler
     }
 #endif
 
+    //! Returns the side of a point w.r.t. a range,
+    //! 1 if it is located left
+    //! -1 if it is located right
+    //! 0 if it is collinear
```

Comment:
This is kind of a generic function, I think it works very nice and indicates better the intentions (actually it could be used in several places in this complex file). I use it now in the new function. I could move it and unit-test it separately.

The only "disadvantage" is that sides are not cached, that means that (especially if we use it more frequently), it might calculate the same side several times. However, solving that would require additional complexity (i.e. pass 3 sides, and it is called four times, so at that place we need to add 12 (a bit less, some are duplicate) side variables.

Please let me know your opinion.

Reply:
It depends whether this is indeed a bottleneck in real use cases. That is, if it is common for an application to calculate several sides and the performance gain is significant then caching sides could worth it.